src/deterministic_vi/elbo_objective.jl

```diff
@@ -127,45 +127,91 @@ function populate_fsm_vecs!{NumType <: Number}(
                              elbo_vars.calculate_derivs,
                              elbo_vars.calculate_hessian,
                              ea.patches, ea.active_sources,
-                             ea.psf_K, ea.num_allowed_sd,
+                             ea.num_allowed_sd,
                              tile_sources, tile, h, w, gal_mcs, star_mcs)
 end
 
 
 """
-Add the contributions of a single source to E_G_s and var_G_s, which are cleared
-and then updated in place.
+Populate fs0m and fs1m for all sources for a given source and pixel.
+
+Args:
+    - elbo_vars: Elbo intermediate values.
+    - ea: Model parameters
+    - fs0m, fs1m: Star and galaxy parameters updated in place.
+    - s: The source to use
+    - b: The band
+    - x: The pixel location in the full image
+    - is_active_source: Whether or not the source is being optimized
+    - gal_mcs: Galaxy components
+    - star_mcs: Star components
+
+Returns:
+    Updates fs0m and fs1m in place with the total
+    shape contributions to this pixel's brightness.
+"""
+function populate_fsm!{NumType <: Number}(
+                    elbo_vars::ElboIntermediateVariables{NumType},
+                    ea::ElboArgs{NumType},
+                    fs0m::SensitiveFloat{StarPosParams, NumType},
+                    fs1m::SensitiveFloat{GalaxyPosParams, NumType},
+                    s::Int,
+                    b::Int,
+                    x::SVector{2,Float64},
+                    is_active_source::Bool,
+                    gal_mcs::Array{GalaxyCacheComponent{NumType}, 4},
+                    star_mcs::Array{BvnComponent{NumType}, 2})
+    # ensure tile.b is a filter band, not an image's index
+    @assert 1 <= b <= B
+    Model.populate_fsm!(elbo_vars.bvn_derivs,
+                        fs0m, fs1m,
+                        elbo_vars.calculate_derivs,
+                        elbo_vars.calculate_hessian,
+                        s, x, is_active_source,
+                        ea.num_allowed_sd,
+                        ea.patches[s, b].wcs_jacobian,
+                        gal_mcs, star_mcs)
+end
+
+
+"""
+Calculate the contributions of a single source for a single pixel to
+the sensitive floats E_G_s and var_G_s, which are cleared and updated in place.
 
 Args:
     - elbo_vars: Elbo intermediate values, with updated fs1m_vec and fs0m_vec.
     - ea: Model parameters
-    - sbs: Source brightnesses
+    - E_G_s, var_G_s: The expectation  and variance of the brightnesses of this
+          source at this pixel, updated in place.
+    - fs0m, fs1m: The star and galaxy shape parameters for this source at
+          this pixel.
+    - sb: Source brightnesse
     - s: The source, in 1:ea.S
     - b: The band
 
 Returns:
-    Updates elbo_vars.E_G_s and elbo_vars.var_G_s in place with the brightness
-    for this sourve at this pixel.
+    Updates E_G_s and var_G_s in place with the brightness
+    for this source at this pixel.
```

Comment:
No, I haven't seen that elsewhere.  I'd be happy to conform to some style guide if we had one.  I do think it's pretty useful for functions to document what variables they change, and I know we disagree about this.  I think the way we've been doing it where I just document functions I care about has been working ok, so if you don't object, I'm just going to keep doing it.